### PR TITLE
dist/constant - synchronize Changes and Makefile.PL into core

### PR DESCRIPTION
**Pull request by demerphq**
This requires a version bump of course, thus then necessitating a new CPAN release.

See https://github.com/Perl/perl5/pull/20867#issuecomment-1448422564 where @haarg got this ball rolling. 

Ideally we should have everything in dist that we need to recreate the release. constant.pm doesnt seem to have a stand alone repo.

Fixes this distribution for https://github.com/Perl/perl5/issues/20874

### CODE REVIEW

File: dist/constant/Makefile.PL [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    PL_FILES        => {},
+    PREREQ_PM       => \%prereq,
+    META_MERGE          => {
+        resources       => {
+            repository  => "git://perl5.git.perl.org/perl.git",
+            license     => 'http://dev.perl.org/licenses/',
+            homepage    => 'https://metacpan.org/module/constant',
+            irc         => 'irc://irc.perl.org/#p5p',
+            mailinglist => 'http://lists.perl.org/list/perl5-porters.html',
+            bugtracker  => "https://rt.perl.org/rt3/Search/Results.html?Query=Queue='perl5' AND Content LIKE 'module=constant' AND (Status='open' OR Status='new' OR Status='stalled')",
```
Comment (tonycoz) [inline dist/constant/Makefile.PL]: Out of date repository and bugtracker URLs.

Comment (demerphq) [reply dist/constant/Makefile.PL]: Fixed, I think. Please scrutinize carefully. I am not sure I did it right.

File: dist/constant/Makefile.PL [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    PL_FILES        => {},
+    PREREQ_PM       => \%prereq,
+    META_MERGE          => {
+        resources       => {
+            repository  => 'https://github.com/Perl/perl5.git',
+            license     => 'http://dev.perl.org/licenses/',
+            homepage    => 'https://metacpan.org/module/constant',
+            irc         => 'irc://irc.perl.org/#p5p',
+            mailinglist => 'http://lists.perl.org/list/perl5-porters.html',
+            bugtracker  => 'https://github.com/Perl/perl5/Issues',
```
Comment (tonycoz) [inline dist/constant/Makefile.PL]: Lower case "I" in issues, this one 404s.

The rest look fine.

It's possible to expand the resources key out by switching to meta-spec 2, (see `cpan/Scalar-List-Utils/Makefile.PL` for example) but I'm mostly trying to pick up obvious errors.  Refactoring to make them all meta-spec 2 seems out of scope for me.

Comment (demerphq) [reply dist/constant/Makefile.PL]: Fixed. Re: meta-spec 2, I think in the future we should look at making these Makefile.PL's more consistent, but I agree it is not in scope now.
